src/main/java/com/thealgorithms/sorts/SmoothSort.java

```diff
@@ -0,0 +1,162 @@
+package com.thealgorithms.sorts;
+
+/**
+ * @author Pankaj Kumar Bind (https://github.com/Pankaj-Bind)
+ * @see SmoothSort Algorithm
+ * Detailed description of this algorithm given below the code
+ * 
+ * SmoothSort uses a different data structure when compared to heap sort. 
+ * Specifically, it uses a data structure called the Leonardo heap, which is a binary tree with the property that the root of each subtree has one fewer element than its parent. 
+ * The shape of the Leonardo heap is determined by a sequence of numbers called the Leonardo.
+ * Define the Leonardo numbers
+ *	Leonardo numbers are a sequence of numbers defined by: L(0) = 1, L(1) = 1,
+ *	L(n) = L(n - 1) + L(n - 2) + 1 for n > 1.
+ * */
+
+import java.util.Arrays;
+
+public class SmoothSort {
+
+	static int leonardo(int k) {
+		if (k < 2) {
+			return 1;
+		}
+		return leonardo(k - 1) + leonardo(k - 2) + 1;
+	}
+
+	/*
+     * Build the Leonardo heap by merging
+	 * pairs of adjacent trees
+     * */
+	static void heapify(int[] arr, int start, int end) {
+		int i = start;
+		int j = 0;
+		int k = 0;
+
+		while (k < end - start + 1) {
+			if ((k & 0xAAAAAAAA) != 0) {
+				j = j + i;
+				i = i >> 1;
+			} else {
+				i = i + j;
+				j = j >> 1;
+			}
+
+			k = k + 1;
+		}
+
+		while (i > 0) {
+			j = j >> 1;
+			k = i + j;
+			while (k < end) {
+				if (arr[k] > arr[k - i]) {
+					break;
+				}
+				int temp = arr[k];
+				arr[k] = arr[k - i];
+				arr[k - i] = temp;
+				k = k + i;
+			}
+
+			i = j;
+		}
+	}
+
+	/*
+     * Smooth Sort function
+     * */
+	static int[] smoothSort(int[] arr) {
+		int n = arr.length;
+
+		int p = n - 1;
+		int q = p;
+		int r = 0;
+
+		/*
+     	* Build the Leonardo heap by merging
+     	* pairs of adjacent trees
+     	* */
+		while (p > 0) {
+			if ((r & 0x03) == 0) {
+				heapify(arr, r, q);
+			}
+
+			if (leonardo(r) == p) {
+				r = r + 1;
+			} else {
+				r = r - 1;
+				q = q - leonardo(r);
+				heapify(arr, r, q);
+				q = r - 1;
+				r = r + 1;
+			}
+
+			int temp = arr[0];
+			arr[0] = arr[p];
+			arr[p] = temp;
+			p = p - 1;
+		}
```

Comment:
This could be moved to a seperate method.

Reply:
issue has been sorted